src/v/kafka/protocol/schemata/generator.py

```diff
@@ -400,6 +400,7 @@
     int16=("int16_t", "read_int16()"),
     int32=("int32_t", "read_int32()"),
     int64=("int64_t", "read_int64()"),
+    float64=("double", "read_double()"),
```

Comment:
As of C++23 we get `std::float64_t` which is probably correct, as long as `std::numeric_limits<std::float64_t>::is_iec559` is also true.

We can't use that (yet), so it may be worth creating a typedef in `kafka/types.h`, with static_assert for size and iec559.

I don't know if this should be `float64_t`, too, but most of the other changes in this commit should probably be `float64_t` instead of `double`.

Reply:
+1 to float64 just so the terminology is aligned with kafka protocol terminology

Reply:
Makes sense, updated